key/key.go

```diff
@@ -44,9 +44,9 @@ type keyImpl struct {
 	key interface{}
 }
 
-// NewKey returns a Key object implementing the Key interface
+// New returns a Key object implementing the Key interface
```

Comment:
First sentence is fine.

I like mentioning the bad data required to panic, since a godoc reader can't see otherwise the panic without a conscious click of the source link.  Up to you on this.